ynr/apps/elections/uk/every_election.py

```diff
@@ -186,6 +186,50 @@ def get_or_create_post_election(self, parent):
             )
         return (self.post_election_object, self.post_election_created)
 
+    def delete_post_election(self):
+        try:
+            pee = PostExtraElection.objects.get(
+                ballot_paper_id=self["election_id"]
+            )
+        except PostExtraElection.DoesNotExist:
+            # if it doesn't already exist, this might be because:
+            # - we deleted it in EE before it was ever imported into YNR
+            # - we already deleted it
+            # either way, we wanted it gone and now its not there
+            return
+
+        pee.delete()
+
+    def delete_election(self):
```

Comment:
Do you think this would be better as a [class?] method on `Election`? There are other places (the admin area at the moment, but we might want to remove that feature) where we might delete elections and want this check to happen. 

I understand the try/except would need to be refactored if it's an instance method, but even moving the other logic (once we have an instance) to a `save_delete` method would be nice.

The same applies (although there isn't any logic on it yet) to `PostExtraElection` / `delete_post_election` above.

Reply:
Yeah. I'll have a look at moving both of these methods over to the Election/PEE classes tomorrow. I think the current behaviour of `/admin` (where it shows you any related objects that would be deleted by cascading the delete and allows you to carry on or abort) is fine - sometimes we actually might want to do that. Also remember that by default we can't edit elections in `/admin` anyway: https://github.com/DemocracyClub/yournextrepresentative/blob/master/ynr/settings/base.py#L418 so we've got to be fairly determined to do it :)

I can see that having a `safe_delete()` method on the model is handy though.